File: geekbang/algorithm/source-code/3/lc94 二叉树的中序遍历.go
func inorderTraversal(root *TreeNode) (res []int) {
	var inorder func(node *TreeNode)
	inorder = func(node *TreeNode) {
		// 终止函数
		if node == nil {
			return
		}
		// 递归逻辑
		inorder(node.Left)
		// 当前层处理
		res = append(res, node.Val)
		// 递归逻辑
		inorder(node.Right)
	}
	inorder(root)
	return
}

File: geekbang/algorithm/source-code/3/lc94 二叉树的中序遍历.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
 var inorderTraversal = function(root) {
    var seq = [];

    var dfs = function(root) {
        if (root == null) return;
        dfs(root.left);
        seq.push(root.val);
        dfs(root.right);
    }

    dfs(root);
    return seq;
}